Update privacy docs to be really explicit about inputs vs text (#6761)
PostHog/posthog.com: contents/docs/session-replay/privacy.mdx

```ts
posthog.init('<ph_project_api_key>', {
    session_recording: {
        // `maskInputFn` only applies to selected elements, so make sure to set this to true if you want this to work globally

>         maskAllInputs: true,

Reviewer: is it maskInputFn or maskAllInputs?

Author: The comment is meant to indicate that `maskInputFn` only runs for inputs that have been marked as needing masking. So `maskAllInputs: true` is required for this case.

Please feel free to make changes to what you think is clearer as the whole motivation of this PR is that some users didnt get it

General text is not masked by default, but we provide multiple options for masking text:

> > **IMPORTANT**: The `text` related config options only apply to non-input text. Inputs are masked differently and as such have separate methods as detailed above
> 
> ### Mask all text

Reviewer: ```suggestion
### Mask all text

> **IMPORTANT**: The `text` related config options only apply to non-input text. Inputs are masked differently and have separate methods (as detailed [above](#input-elements)).
```

```ts
posthog.init('<ph_project_api_key>', {
    session_recording: {
        maskTextSelector: "*" // Masks all text elements (not including inputs)
    }
})
```

### Mask or un-mask specific text

You can use a CSS selector to mask specific elements. For example, you may want to mask all elements with the class `email` or the ID `sensitive`.

```ts
posthog.init('<ph_project_api_key>', {
    session_recording: {

>         // REMINDER - This does not apply to input elements.
>         maskTextSelector: ".email, #sensitive" // masks all elements with the class "email" or the ID "sensitive"

Reviewer: ```suggestion
        maskTextSelector: ".email, #sensitive" // masks all elements with the class "email" or the ID "sensitive". This does not apply to input elements.

```

```ts
posthog.init('<ph_project_api_key>', {
    session_recording: {

>         // `maskTextFn` only applies to selected elements, so make sure to set to "*" if you want this to work globally
>         // REMINDER - This does not apply to input elements.

Reviewer: ```suggestion
        // `maskTextFn` only applies to selected elements, so make sure to set to "*" if you want this to work globally.
        // This does not apply to input elements.
```

<div class="ph-no-capture">I won't be captured at all!</div>
```

Note that adding `ph-no-capture` will also prevent any [autocapture](/docs/data/autocapture) events from being captured from that element.

> 

Reviewer: ```suggestion
```

> # Common example configs

Reviewer: For some reason this is not shown in the preview. https://posthog-36hnvs1z2-post-hog.vercel.app/docs/session-replay/privacy

Perhaps because it should be H2 and not H1?
```suggestion
## Common example configs
```

> ## Maximum privacy - mask everything

Reviewer: ```suggestion
### Maximum privacy - mask everything
```

```
{
    maskAllInputs: true,
    maskTextSelector: "*"
}
```

> ## Good privacy - try to mask all email / password related things

Reviewer: ```suggestion
## Masking email and password
```

Author: I'd really like to keep the original wording. In particular because the example does not actually guarantee emails will be masked, it is just best effort

> To strike a balance of usefulness versus privacy friendly, you can choose to only mask content that looks like it contains an email or password.
> With passwords, it is important to note that typically the "click to show password" buttons, turn the input type to `text` which would then reveal the value. If we check for something else like the `id` field then we can ensure that it is never captured (you of course need to ensure this field is added)

Reviewer: ```suggestion
You can mask content that looks like it contains an email or password.

For passwords, it is important to note that "click to show password" buttons typically turn the input type to `text`. This would then reveal the password. Thus instead of checking for `type='password'`, you need to check a different field, like `id`:
```
